test/DynamoCoreWpfTests/PackageManagerUITests.cs

```diff
@@ -91,6 +95,39 @@ public void PackagePublishWindowClosesWithDynamo()
 
         }
 
+        [Test]
+        public void AdddingaPackageRaisesCanExecuteChangeOnDelegateCommand()
+        {
+            //create a new synchronization context since the package searchview from previous UI tests is still
+            //running and throws exception when attempting to modify searchresults
+            SynchronizationContext.SetSynchronizationContext(new SynchronizationContext());
+            var vm = new PublishPackageViewModel(ViewModel);
+            ViewModel.OnRequestPackagePublishDialog(vm);
+
+            //find a customnode to add to the package
+            string packagedirectory = Path.Combine(GetTestDirectory(ExecutingDirectory), "pkgs");
+            var packages = Directory.EnumerateDirectories(packagedirectory);
+            var first = Path.GetFullPath(packages.First());
+            string dyfpath = Path.Combine(first, "dyf");
+            var customnodes = Directory.GetFiles(dyfpath);
+            var firstnode = customnodes.First();
+                      
+            var canExecuteChangedFired = 0;
+            vm.SubmitCommand.CanExecuteChanged += ((o, e) => { canExecuteChangedFired ++; });
```

Comment:
If the BeginInvoke you use to set `CanExecuteChanged` is required, I worry that having a handler for `CanExecuteChanged` here won't be called while the test is still running (it's being invoked asynchronously). I know you use `DoEvents` later on, which I'm also not a big fan of because it's really difficult for the person reading this test to understand what is happening and in what order. It might be a necessary evil. Overall, this might cause some flaky behavior with this test.